event_sale_session/models/sale_order_line.py

```diff
@@ -13,28 +13,33 @@ class SaleOrderLine(models.Model):
     event_session_id = fields.Many2one(
         string="Session",
         comodel_name="event.session",
+        compute="_compute_event_session_id",
         domain="[('event_id', '=', event_id)]",
+        store=True,
+        readonly=False,
+        precompute=True,
     )
 
-    @api.onchange("event_id")
-    def _onchange_event_id(self):
-        # OVERRIDE to automatically set the session, if there's only one available
-        # and also to clear event_session_id if it's inconsistent with the event
-        event = self.event_id
-        if event.session_count == 1:
-            self.event_session_id = event.session_ids[0]
-        elif not event.use_sessions or event != self.event_session_id.event_id:
-            self.event_session_id = False
-        return super()._onchange_event_id()
+    @api.depends("event_id")
+    def _compute_event_session_id(self):
+        event_lines = self.filtered("event_id")
+        (self - event_lines).event_session_id = False
+        for line in event_lines:
+            if line.event_id != line.event_session_id.event_id:
```

Comment:
```suggestion
            if not line.event_id.use_sessions or line.event_id != line.event_session_id.event_id:
```
I think that you need to preserve the fist part of the condition

Reply:
Good catch, thanks.